### PR TITLE
Editorial workflow

### LINKED ISSUE #205: Editorial Workflow: Bug in persistence if branch already exists
When using the editorial workflow, trying to edit a content that already has a branch, it fails on save because it tries to create a new branch and fails.

When trying to edit a content already under editorial workflow, the CMS should simply commit to the  changes to the existing branch.

### LINKED ISSUE #208: Editorial Workflow: Merge conflicts should be resolved automatically
We already track all changed files in the metadata so we could resolve the conflicts automatically by overwriting.

### PR DESCRIPTION
**Pull request by cassiozen**
**- Summary**

Refactor on Editorial Workflow. Main objectives are:

* Make "unpublished entries" loading/persistence method signatures more similar to regular entries ones.
* Singular route for new entries, first edits and edits on entries already under EW.
* Always verify existence of previously edits under EW when trying to edit (closes #205)
* On further edits of a draft entry, update the last commit sha on the metadata
* Automatic merge by overwriting (closes #208)

**- Description for the changelog**
Refactor on Editorial Workflow, with added support for automatic PR merge by overwriting.

**- A picture of a cute animal (not mandatory but encouraged)**
ʕ •ᴥ•ʔ

### CODE REVIEW

File: src/backends/github/API.js [hunk 1]
```diff
@@ -335,14 +359,38 @@ export default class API {
...
+  }
+
+  forceMergePR(pullrequest, objects) {
+    const files = objects.files.concat(objects.entry);
+    const fileTree = this.composeFileTree(files);
+    let commitMessage = "Automatically generated. Merged on Netlify CMS\n\nForce merge of:";
+    files.forEach((file) => {
+      commitMessage += `\n* "${ file.path }"`;
     });
+    console.log("%c Automatic merge not possible - Forcing merge.", "line-height: 30px;text-align: center;font-weight: bold"); // eslint-disable-line
```
Comment (calavera) [inline src/backends/github/API.js]: do we need this log?

File: src/valueObjects/errors/index.js [hunk 2]
```diff
@@ -0,0 +1,3 @@
+export { default as APIError } from './APIError';
+export { default as EditorialWorkflowError } from './EditorialWorkflowError';
+
```
Comment (calavera) [inline src/valueObjects/errors/index.js]: remove line?

File: src/backends/github/API.js (hunk 1 above)
Comment (cassiozen) [inline src/backends/github/API.js]: Great question: There's no way to hide a 404 (or 409 in this case) from Chrome's console. So I log this type of messages before and after a potential (and expected) non-20x response.

File: src/valueObjects/errors/index.js (hunk 2 above)
Comment (cassiozen) [inline src/valueObjects/errors/index.js]: Line 3? Current lint config requires a newline at end of file...

File: src/backends/github/API.js (hunk 1 above)
Comment (cassiozen) [inline src/backends/github/API.js]: Do you think this is stupid? I can remove them if you prefer.

File: src/valueObjects/errors/index.js (hunk 2 above)
Comment (calavera) [inline src/valueObjects/errors/index.js]: ok

File: src/backends/github/API.js (hunk 1 above)
Comment (calavera) [inline src/backends/github/API.js]: interesting, sounds good

Comment (biilmann) [inline src/backends/github/API.js]: I would log them with `console.error` so we know they're supposed to be there if we just grep for `console.log`

**Pull request merged**